test/test_sparse_csr.py

```diff
@@ -3353,6 +3353,45 @@ def setUp(self):
 @skipIfNoTriton
 class TestSparseCompressedTritonKernels(TestCase):
 
+    @onlyCUDA
+    @skipIfRocm
+    @dtypes(torch.half, torch.bfloat16, torch.float)
+    @dtypesIfCUDA(torch.half, *[torch.bfloat16] if SM80OrLater else [], torch.float)
+    @unittest.skipIf(IS_FBCODE and IS_REMOTE_GPU, "Test requires Triton")
+    def test_triton_bsr_softmax(self, device, dtype):
+        from functools import partial
+        from torch.sparse._triton_ops import bsr_softmax, tile_to_blocksize
+
+        tensor = partial(make_tensor, device=device, dtype=dtype, low=1.0, high=3.0)
+
+        # NOTE: batch dims with zero sizes are not supported in `to_sparse_bsr`.
+        batches = [(), (2,), (2, 2)]
+        size = [6, 12, 0]
+        block_size = [2, 3]
+
+        for row_block, col_block, b, m, n in itertools.product(block_size, block_size, batches, size, size):
+            input = tensor(b + (m, n))
+            input.diagonal(dim1=-2, dim2=-1).fill_(m * n)
+
+            input_tiled = tile_to_blocksize(input, (row_block, col_block))
+            input_tiled = input_tiled.moveaxis(-4, -1).moveaxis(-4, -1)
+            if m // row_block > n // col_block:
+                input_tiled.tril_()
+            else:
+                input_tiled.triu_()
+
+            bsr = input.to_sparse_bsr((row_block, col_block))
+
+            if m // row_block > n // col_block:
+                dense = -100 * (input_tiled == 0.) + input_tiled.tril()
+            else:
+                dense = -100 * (input_tiled == 0.) + input_tiled.triu()
+            input_tiled.copy_(dense)
+
+            res_tri = bsr_softmax(bsr)
+            res_dense = input.softmax(-1)
```

Comment:
I'd expect the reference to be https://pytorch.org/docs/master/generated/torch.sparse.softmax.html#torch.sparse.softmax since we're really following masked semantics here if this softmax is to be used in the context of SDPA.

Is that correct?

Reply:
`torch.sparse` is limited to non-batched COO only. Besides, the dense reference is masked to produce zeroes after softmax.